var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -109,13 +109,13 @@ class VtkM(CMakePackage, CudaPackage, ROCmPackage):
     # may not be used for VTK-m depending on the default selected by Kokkos
     depends_on("kokkos +rocm", when="+kokkos +rocm")
     # Propagate AMD GPU target to kokkos for +rocm
-    for amdgpu_value in ROCmPackage.amdgpu_targets:
-        depends_on(
-            "kokkos amdgpu_target=%s" % amdgpu_value,
-            when="+kokkos +rocm amdgpu_target=%s" % amdgpu_value,
-        )
+    # for amdgpu_value in ROCmPackage.amdgpu_targets:
+    #     depends_on(
+    #         "kokkos amdgpu_target=%s" % amdgpu_value,
+    #         when="+kokkos +rocm amdgpu_target=%s" % amdgpu_value,
+    #     )
```

Comment:
Is there another mechanism that enforces that the `kokkos` dependency will be built for all `amdgpu_target`s requested for this package?